### ISSUE TITLE
pipe.scheduler.sigmas change after forward call

### ISSUE THREAD

**Issue by neuron-party** [huggingface/diffusers#7904]
### Describe the bug

I noticed this issue when trying to manually modify the sigmas in my pipeline scheduler. Despite changing the values in this tensor, the sigmas tensor shrinks to the number of inference steps after one inference instance. Subsequent calls no longer change the values of the sigmas. 

### Reproduction

pipe.scheduler.sigmas = pipe.scheduler.sigmas * 4 # manually modifying sigmas
_ = pipe(..., num_inference_steps = N)
print(pipe.scheduler.sigmas) # len is N + 1, values do not reflect the manual modifications and no longer change in subsequent calls

### System Info

diffusers==0.27.2

**Comment by yiyixuxu** [comment]
hi!
if you are looking to use custom sigmas, you can pass it down as `sigmas` from your pipeline 
see doc https://huggingface.co/docs/diffusers/main/en/using-diffusers/schedulers#custom-timestep-schedules
and this PR https://github.com/huggingface/diffusers/pull/7817

you will need to install diffusers from the source and use the most recent updated version

**Comment by yiyixuxu** [comment]
also `simgas` change after forward call is intended behavior because we reset the timesteps based on the `num_inference_steps` argument

**Comment by neuron-party** [comment]
Great thanks! I noticed this functionality still isn't supported for SDXL ControlNets, specifically diffusers.StableDiffusionXLControlNet pipeline, and I was wondering if you will be adding this functionality for this pipeline? Thanks!

**Comment by yiyixuxu** [comment]
yes - do you want to open a PR to help us add this to the Controlnet pipeline?
if not we can ask other members of the community for help too:)

**Comment by neuron-party** [comment]
Sure! Will work on it over this weekend

**Comment by neuron-party** [comment]
PR: https://github.com/huggingface/diffusers/pull/7913
